clmm/theory/ccl_v3.py

```diff
@@ -0,0 +1,180 @@
+            'critical': 'critical',
+            'virial': 'critical'}
+        self.hdpm_dict = {'nfw': ccl.halos.HaloProfileNFW,
+                          'einasto': ccl.halos.HaloProfileEinasto,
+                          'hernquist': ccl.halos.HaloProfileHernquist}
+        self.cosmo_class = CCLCosmology
+        self.hdpm_opts = {'nfw': {'truncated': False,
+                                  'projected_analytic': True,
+                                  'cumul2d_analytic': True},
+                          'einasto': {'truncated': False},
+                          'hernquist': {'truncated': False,
+                                        'projected_analytic': True,
+                                        'cumul2d_analytic': True}}
+        self.cor_factor = _patch_rho_crit_to_cd2018(ccl.physical_constants.RHO_CRITICAL)
+        self.__mdelta_cor = 0.0 ## mass with corretion for input
+        #self.hdpm_opts['einasto'].update({'alpha': 0.25}) # same as NC default
+
+        # Set halo profile and cosmology
+        self.mdef = None
+        self.conc = None
+        self.set_halo_density_profile(halo_profile_model, massdef, delta_mdef)
+        self.set_cosmo(None)
+
+
+    # Functions implemented by child class
+
+
+    def _update_halo_density_profile(self):
+        """"updates halo density profile with set internal properties"""
+        # prepare mdef object
+        self.mdef = ccl.halos.MassDef(self.delta_mdef, self.mdef_dict[self.massdef])
+        # setting concentration (also updates hdpm)
+        self.cdelta = self.cdelta if self.hdpm else 4.0 # ccl always needs an input concentration
+
+    def _get_concentration(self):
+        """"get concentration"""
+        return self.conc.c
+
+    def _get_mass(self):
+        """"get mass"""
+        return self.__mdelta_cor*self.cor_factor
+
+    def _set_concentration(self, cdelta):
+        """"set concentration. Also sets/updates hdpm"""
+        self.conc = ccl.halos.ConcentrationConstant(c=cdelta, mdef=self.mdef)
+        with ccl.UnlockInstance(self.mdef):
+            self.mdef.concentration = self.conc
+        self.hdpm = self.hdpm_dict[self.halo_profile_model](
+            self.conc, **self.hdpm_opts[self.halo_profile_model])
+        self.hdpm.update_precision_fftlog(
+            padding_lo_fftlog=1e-4, padding_hi_fftlog=1e3)
+
+    def _set_mass(self, mdelta):
+        """" set mass"""
+        self.__mdelta_cor = mdelta/self.cor_factor
+
+    def _set_einasto_alpha(self, alpha):
+        if alpha is None:
+            self.hdpm.update_parameters(alpha='cosmo')
+        else:
+            self.hdpm.update_parameters(alpha=alpha)
+
+    def _get_einasto_alpha(self, z_cl=None):
+        """"get the value of the Einasto slope"""
+        if self.hdpm.alpha!='cosmo':
+            a_cl = 1 # a_cl does not matter in this case
+        else:
+            a_cl = self.cosmo.get_a_from_z(z_cl)
+        return self.hdpm._get_alpha(self.cosmo.be_cosmo, self.__mdelta_cor, a_cl)
```

Comment:
This line is different from that of v2. In v3 `self.mdef` is not needed for `_get_alpha`.